src/test/tcp/test_tcp.c

```diff
@@ -49,6 +52,45 @@ static void _mylog(const char* fileName, const int lineNum, const char* funcName
     fflush(stdout);
 }
 
+static void queue_send_u16(const char* queuename, uint16_t i) {
+    mqd_t mq = mq_open(queuename, O_WRONLY|O_CREAT, 0644, NULL);
+    g_assert(mq != (mqd_t)-1);
+
+    // convert i in string before sending it
+    char buf[10] = {0};
+    int len = snprintf(buf, 10, "%d", i);
+    mq_send(mq, buf, len, 10);
+
+    mq_close(mq);
+}
+
+static short queue_recv_u16(const char* queuename) {
+    mqd_t mq;
+    guint64 val;
+    GError *error = NULL;
+    struct mq_attr attr;
+    unsigned int priorite;
+    char buf[10] = {0};
+
+    // let the server write in the queue
+    usleep(1000);
```

Comment:
Sleeping in tests is almost never the right solution for resolving race conditions. It makes the test slower, and still leaves us with a race condition.

We should find a way to properly synchronize (such as by creating the ipc mechanism before we kick off the tests, as mentioned above)

Reply:
Yes, this was a first step. mqueue allows to do async and cleans this stuff. I don't know if I will be able to do the same thing with system v, what do you think ? But maybe the problem that I encountered came from the Posix implementation and I will not get it with system v ?

Reply:
I'm fine with sticking with posix mqueues now that I know how to manipulate them from the command-line. It's also how we solve this problem: in the test command in the CMakeLists.txt file, we should delete the queue if it already exists, create a fresh queue, and *then* start the test.

That way when we get here the queue is guaranteed to already exist. The other side might not have written to it yet, but that's ok - the read will just block until there's a message there, which is what we want.